src/vivarium_gates_nutrition_optimization_child/components/observers.py

```diff
@@ -9,77 +9,96 @@
+    results production and have this component manage adjustments to the
+    final column labels for the subgroups.
+    """
+
+    def register_stratifications(self, builder: Builder) -> None:
+        """Register each desired stratification with calls to _setup_stratification"""
+        super().register_stratifications(builder)
+
+        builder.results.register_stratification(
+            "wasting_state",
+            [category.value for category in data_keys.CGFCategories],
+            self.child_growth_risk_factor_stratification_mapper,
+            is_vectorized=True,
+            requires_values=["child_wasting.exposure"],
+        )
+        builder.results.register_stratification(
+            "stunting_state",
+            [category.value for category in data_keys.CGFCategories],
+            self.child_growth_risk_factor_stratification_mapper,
+            is_vectorized=True,
+            requires_values=["child_stunting.exposure"],
+        )
+        builder.results.register_stratification(
+            "maternal_supplementation",
+            results.MATERNAL_SUPPLEMENTATION_TYPES,
+            is_vectorized=True,
+            requires_columns=["maternal_supplementation_exposure"],
+        )
+        builder.results.register_stratification(
+            "iv_iron",
+            results.DICHOTOMOUS_COVERAGE_STATES,
+            is_vectorized=True,
+            requires_columns=["iv_iron_exposure"],
+        )
+        builder.results.register_stratification(
+            "bmi_anemia",
+            ["cat4", "cat3", "cat2", "cat1"],
+            is_vectorized=True,
+            requires_columns=["maternal_bmi_anemia_exposure"],
+        )
+
+    ###########################
+    # Stratifications Details #
+    ###########################
+
+    # noinspection PyMethodMayBeStatic
+    def child_growth_risk_factor_stratification_mapper(self, pop: pd.DataFrame) -> pd.Series:
+        # applicable to stunting and wasting
+        mapper = {
+            "cat4": data_keys.CGFCategories.UNEXPOSED.value,
+            "cat3": data_keys.CGFCategories.MILD.value,
+            "cat2": data_keys.CGFCategories.MODERATE.value,
+            "cat1": data_keys.CGFCategories.SEVERE.value,
+        }
+        return pop.squeeze(axis=1).replace(mapper)
+
+    def map_age_groups(self, pop: pd.DataFrame) -> pd.Series:
+        """Map age with age group name strings
+
+        Parameters
+        ----------
+        pop
+            A DataFrame with one column, an age to be mapped to an age group name string
+
+        Returns
+        ------
+        pandas.Series
+            A pd.Series with age group name string corresponding to the pop passed into the function
+        """
+        bins = self.age_bins["age_start"].to_list() + [self.age_bins["age_end"].iloc[-1]]
+        labels = self.age_bins["age_group_name"].to_list()
+        # need to include lowest to map people who are exactly 0
+        age_group = pd.cut(
```

Comment:
I tried initializing simulants at the age of half a time step in the fertility component but they were still being born at age 0.0. In the interest of time I went ahead with this solution since it's working.

Reply:
Having read the docs for `pd.cut` this is actually just the correct solution. Sorry for the confusion I introduced on Friday.